core/base/timer.cpp

```diff
@@ -96,7 +96,7 @@ void CpuTimer::wait(time_point& time) {}
 std::chrono::nanoseconds CpuTimer::difference_async(const time_point& start,
                                                     const time_point& stop)
 {
-    return std::chrono::duration_cast<std::chrono::nanoseconds, int64>(
+    return std::chrono::duration_cast<std::chrono::nanoseconds>(
         stop.data_.chrono - start.data_.chrono);
```

Comment:
Just to ensure I got it correctly.
duration_cast signature:
```
template< class ToDuration, class Rep, class Period >
constexpr ToDuration duration_cast( const std::chrono::duration<Rep,Period>& d );
```
The issue is from type definition of `std::chrono::nanoseconds`
`std::chrono::duration</* int64 */, std::nano>` standard only require it at least 64 bit signed integer.
Thus, it might be not int64.
The type also affect `.count()` return type. 
Could you check that the `.count()` has static_cast to desired type if it is the case?

Reply:
The issue is basically that I used to return int64 before and now return nanoseconds, and I didn't change the function call. It only worked accidentally.

Reply:
I checked, we only use `.count()` in places relevant to this issue where it will be cast to `double`, so no problems expected